### DISCUSSION TITLE
404 return for every update request in livewire.

### DISCUSSION THREAD

**Discussion by taha7672** [livewire/livewire#6960] [1. Help] [answered]
I have put the Laravel project in a subfolder (lmapnext) and use server.php and .htaccess in root of laravel  to  skip public from URL. Laravel is working fine but in Livewire first pages of the livewire component works fine (those whose routes are defined in web.php) other update post request didn't work (URL is example.com/lmapnext/livewire/update),
For Example call a method from wire:click method doest work, 
The project works fine in PHP artisan serve method ,

I am using PHP 8.1, livewire3. Actually want to put Laravel project in core PHP project for this I am using subfolder.

`
in HandleRequests add /lmapnext
lmapnext\vendor\livewire\livewire\src\Mechanisms\HandleRequests\HandleRequests.php

function boot()
    {
        app($this::class)->setUpdateRoute(function ($handle) {
            return Route::post('/lmapnext/livewire/update', $handle)->middleware('web');
        });

        $this->skipRequestPayloadTamperingMiddleware();
    }
`

`

in config/livewire 
   'asset_url'  => '/lmapnext/livewire/livewire.js',

`
`

.htaccess
<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews
    </IfModule>

    RewriteEngine On

    # Handle Authorization Header
    RewriteCond %{HTTP:Authorization} .
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

    # Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} -d [OR]
    RewriteCond %{REQUEST_FILENAME} -f
    RewriteRule ^ ^$1 [N]

    RewriteCond %{REQUEST_URI} (\.\w+$) [NC]
    RewriteRule ^(.*)$ public/$1 

    # Handle Front Controller...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ server.php
</IfModule>

`
how can i solve this issue i have to use the subfolder method

**Comment by taha7672** [comment]
solve the issue by creating a provider.

path after domain name is PATH_TO_LARAVEl

`

public function boot() { Livewire::setUpdateRoute(function ($handle) { return Route::post(config('constants.PATH_TO_LARAVEl') . '/livewire/update', $handle); }); }

`

**Comment by fabiorcamargo** [comment]
If you are using NGINX, I recently had problems running it. Among everything I tried, what worked for me was adapting the model available in the Laravel documentation.
https://laravel.com/docs/10.x/deployment#nginx

But in your case it seems to have more to do with 'asset_url', change the values and see in your browser's console or network where it is pointing.
https://laravel-livewire.com/docs/2.x/installation#configuring-the-asset-base-url

**Comment by janglapuk** [comment, marked as answer]
That should be configure in `routes/web.php`:

```php
// ...
Livewire::setScriptRoute(function($handle) {
    return Route::get('/lmapnext/livewire/livewire.js', $handle);
});

Livewire::setUpdateRoute(function($handle) {
    return Route::get('/lmapnext/livewire/update', $handle);
});
// ...
```

Results:
```html
...
<script src="/lmapnext/livewire/livewire.js?id=xxx"
        data-csrf="xxx" 
        data-update-uri="/lmapnext/livewire/update" 
        data-navigate-once="true">
</script>
...
```

**Reply by daveharrisonbcc** [reply]
So simple, it took me too long to find a fix for this, thankyou!
